client/src/test/scala/org/http4s/armeria/client/ArmeriaClientSuite.scala

```diff
@@ -205,4 +217,17 @@ class ArmeriaClientSuite extends CatsEffectSuite {
       .reduce(_ + " " + _)
     assertEquals(response, "1! 2! 3! 4! 5!")
   }
+
+  test("timeout response") {
```

Comment:
That's odd but using `Async#fromCompletableFuture` in the client leads to [test failure](https://github.com/http4s/http4s-armeria/actions/runs/5038928682/jobs/9036737918#step:12:254). And vice versa, when using self-written [`fromCompletableFuture`](https://github.com/http4s/http4s-armeria/pull/335/files#diff-120cfbb425cb654ce64648b18cfc564500cb9ffe87d04a03d432e824cbcbc807R106) [all good](https://github.com/http4s/http4s-armeria/actions/runs/5038944727/jobs/9036764047?pr=335).

Reply:
I don't want to say that CE's implementation has bugs, but I neither have ideas about what the heck is going on here 😞 That's for actual CE-ninjas

Reply:
Are you sure if this line is returning `true` when you `cancel` it?

https://github.com/http4s/http4s-armeria/blob/70f4afa48e6b585f812496fb9b300fb14809e08f/client/src/main/scala/org/http4s/armeria/client/ArmeriaClient.scala#L120

```
Returns:
true if this task is now cancelled
```
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#cancel-boolean-

If it is returning `false`, then it is not canceled, so the CE implementation will continue to wait for it to complete.